src/stores/index.js

```diff
@@ -929,40 +834,120 @@ const Store = Reflux.createStore({
    *
    * @param {Object} connection - The current connection.
    */
-  _connect(connection) {
+  async _connect(connection) {
+    // Set the connection's app name to the electron app name of Compass.
+    connection.appname = electron.remote.app.getName();
+
     this.dataService = new DataService(connection);
```

Comment:
Is there any sort of way we could make it less 'side-effect' prone and only assign `this.dataService` and emit `'data-service-initialized'` once the connection happened? What happens if the connection fails?

Reply:
Good thinking. On a brief scan through the code I think there are a number of places which are listening for this event which might make some assumptions about how its being used:
https://github.com/mongodb-js/compass-deployment-awareness/blob/master/src/stores/index.js#L33
https://github.com/mongodb-js/compass-instance-header/blob/master/src/stores/store.js#L15
https://github.com/mongodb-js/compass-sidebar/blob/master/src/stores/store.js#L24
I think those could cause some bugs with this current implementation. - like you said they could have side effects.
I need to investigate if the `topologyDescriptionChanged` is something we need to listen to before we connect. I might differ this to the ticket this cleanup pr is preceding: https://jira.mongodb.org/browse/COMPASS-4393 . How does that sound?

Reply:
Yep, I would even create a separate task for this.

Reply:
Isn't it what happens in the extended model?

https://github.com/mongodb-js/connection-model/blob/master/lib/extended-model.js#L15

Reply:
Created https://jira.mongodb.org/browse/COMPASS-4568 for this